### DISCUSSION TITLE
Is there a way to use pseudo elements like: ::-webkit-slider-thumb?

### DISCUSSION THREAD

**Discussion by stefann01** [tailwindlabs/tailwindcss#6972] [Help]
I try to customize an input type range using tailwind. However, I don't see in the documentation listed pseudo-elements like webkit-slider-thumb or ::-webkit-slider-runnable-track

**Comment by ThatGuySam** [comment]
Theoretically, you should be able to do this with arbitrary variants: 
https://tailwindcss.com/docs/hover-focus-and-other-states#using-arbitrary-variants

```html
<input class="[&::-webkit-slider-thumb]:bg-blue" type="range">
```

You may also be able to support multiple browsers by adding it as a variant in your Tailwind config:
https://tailwindcss.com/docs/plugins#adding-variants
```js
// taillwind.config.js
const plugin = require('tailwindcss/plugin')

module.exports = {
  // ...
  plugins: [
    plugin(function({ addVariant }) {
      addVariant('slider-thumb', ['&::-webkit-slider-thumb', '&::slider-thumb'])
    })
  ]
}
```

```html
<input class="slider-thumb:bg-blue" type="range">
```

**Reply by Garrett-R** [reply]
> Theoretically, you should be able to do this with arbitrary variant

That seems [not to be working](https://play.tailwindcss.com/E5yVqce1Sl).

**Reply by wongjn** [reply]
Arbitrary variant support is v3+, and also `bg-pink` is not a valid color in the configuration, try a valid color like `bg-pink-500`. [Here is a Tailwind Play](https://play.tailwindcss.com/EeEV8IQQKJ) with the fixes (open the *Generated CSS* panel to see the generated class).

**Reply by Garrett-R** [reply]
Thank you!

Ok, I have a complete working example.  It seems you need all these classes (except the `bg-gray-100` which I added for convenience) to get a circular thumb slider working:

```html
<input class="appearance-none bg-gray-100 [&::-webkit-slider-thumb]:bg-pink-500 [&::-webkit-slider-thumb]:appearance-none [&::-webkit-slider-thumb]:w-5  [&::-webkit-slider-thumb]:h-5 [&::-webkit-slider-thumb]:rounded-full" type="range" />
```

![image](https://user-images.githubusercontent.com/6614695/215808890-5cdffa29-4156-4216-aee5-1e779a8c2ae5.png)


([TW Play](https://play.tailwindcss.com/WHY745WyDn))

and for Firefox, you'd duplicate the arbitrary classes and use `-moz-range-thumb`.

**Reply by Jervx** [reply]
> Thank you!
> 
> Ok, I have a complete working example. It seems you need all these classes (except the `bg-gray-100` which I added for convenience) to get a circular thumb slider working:
> 
> ```
> <input class="appearance-none bg-gray-100 [&::-webkit-slider-thumb]:bg-pink-500 [&::-webkit-slider-thumb]:appearance-none [&::-webkit-slider-thumb]:w-5  [&::-webkit-slider-thumb]:h-5 [&::-webkit-slider-thumb]:rounded-full" type="range" />
> ```
> 
> ![image](https://user-images.githubusercontent.com/6614695/215808890-5cdffa29-4156-4216-aee5-1e779a8c2ae5.png)
> 
> ([TW Play](https://play.tailwindcss.com/WHY745WyDn))
> 
> and for Firefox, you'd duplicate the arbitrary classes and use `-moz-range-thumb`.

This works on me ```[&::-webkit-scrollbar]:h-2```. Thank you

![image](https://github.com/user-attachments/assets/835a3bbf-41e0-4a15-a7a2-fb5a798dac9c)

```[&::-webkit-scrollbar]:h-1```

![image](https://github.com/user-attachments/assets/2b61fb0e-aed5-4a8a-9024-8ee795151d82)
